geps/gep-2643/index.md

```diff
@@ -0,0 +1,514 @@
+| A TLSRoute with an IP on its hostname MUST be rejected | Condition on the TLSRoute parent of type `Accepted=False` with Reason `UnsupportedValue`. No request test is required, as the Listener will not be accepted | SupportTLSRoute |
+| A Gateway containing a Listener of type TLS/Terminate SHOULD be accepted, and SHOULD direct the requests to the right TLSRoute when TLSRoute termination is supported. <br/> Issue: [Termination] | Being able to do a request to a TLS route being terminated on gateway (eg.: terminated.example.tld/xpto) | SupportTLSRoute + SupportTLSRouteTermination |
+| A Gateway containing a Listener of type TLS/Passthrough and a Listener of type TLS/Terminate SHOULD be accepted, and should direct the requests to the right TLSRoute when mixed mode is supported. <br/> Issue: [Termination] | Being able to do a request to a TLS route being terminated on gateway (eg.: terminated.example.tld/xpto) and to a TLS Passthrough route on the same gateway, but different host (passthrough.example.tld) | SupportTLSRoute + SupportTLSRouteTermination + SupportTLSRouteMixedMode |
+| A Gateway containing a Listener of type TLS/Passthrough and a Listener of type TLS/Terminate SHOULD NOT be accepted when mixed mode is not supported | Listener condition SHOULD have Condition Conflicted=True with reason `ProtocolConflict`. A request to any route attached to any of the listeners MUST fail. | SupportTLSRoute + SupportTLSRouteTermination + !SupportTLSRouteMixedMode |
+| A Gateway with \*.example.tld on a TLS listener should allow a TLSRoute with hostname some.example.tld to be attached to it (and the same, but with a non wildcard hostname) <br/> Issue: [TLSRoute conformance] | TLSRoute should be able to attach to the Gateway using the matching hostname, a request should succeed | SupportTLSRoute  |
+| Expose support for TLSRoute termination <br/> Issue: [Termination] | For a [Listener](https://gateway-api.sigs.k8s.io/reference/spec/#listener) setting mode: "terminate", TLSRoute should be present in [ListenerStatus.SupportedKinds](https://gateway-api.sigs.k8s.io/reference/spec/#listenerstatus) in case TLSRoute termination is supported | SupportTLSRouteTermination |
+| Explicitly expose that TLSRoute termination is not supported. <br/> Issue: [Termination] | For a [Listener](https://gateway-api.sigs.k8s.io/reference/spec/#listener) setting mode: "terminate" and not being supported, a Listener entry MUST exist but TLSRoute MUST NOT be present in [ListenerStatus.SupportedKinds](https://gateway-api.sigs.k8s.io/reference/spec/#listenerstatus). | SupportTLSRoute + !SupportTLSRouteTermination |
```

Comment:
Can we dictate how the lack of support is implemented?  What if the implementation decides to just reject the Listener?  Otherwise, I don't recall seeing this directive about SupportedKinds mentioned earlier.

Reply:
discussed with @rostislavbobo and fixed both